tools/android/install_base.sh

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env bash
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2024, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Forked from https://github.com/ARM-software/lisa/blob/main/install_base.sh
+#
+
+# shellcheck disable=SC2317
+
+# Read standard /etc/os-release file and extract the needed field lsb_release
+# binary is not installed on all distro, but that file is found pretty much
+# anywhere.
+read_os_release() {
+    local field_name=${1}
+    # shellcheck source=/etc/os-release
+    (source /etc/os-release &> /dev/null && printf "%s" "${!field_name}")
+}
+
+# Test the value of a field in /etc/os-release
+test_os_release() {
+    local field_name=${1}
+    local value=${2}
+
+    if [[ "$(read_os_release "${field_name}")" == "${value}" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+function set_host_arch
+{
+	# Google ABI type for Arm platforms
+	HOST_ARCH="arm64-v8a"
+
+	local machine
+	machine=$(uname -m)
+	if [[ "${machine}" == "x86"* ]]; then
+		HOST_ARCH=${machine}
```

Comment:
I don't think we need any global mutable state here, with the problem that would ensue (someone not calling the function). The function can just `print "%s" "$arch"`  the arch value and code that needs it can call the function as `$(get_android_sdk_host_arch)`, like in basically any language.